### PR TITLE
B-03515 Add dashboard endpoint and storage to hosted happs endpoint

### PR DESCRIPTION
**Pull request by robbiecarlton**

### CODE REVIEW

File: overlays/holo-nixpkgs/hpos-holochain-api/tests/index.test.js [hunk 1]
```diff
@@ -26,7 +26,6 @@ test('Test holochain-api endpoint ', async () => {
   expect(listOfHappsResponse.body[0].sourceChains).toBeFalsy()
```
Comment (JettTech) [inline overlays/holo-nixpkgs/hpos-holochain-api/tests/index.test.js:26]: Can you add a test for the new `/dashboard` endpoint?

**Review by JettTech — CHANGES_REQUESTED** [review]
Did a first pass and have a couple of minor questions and a request.

File: overlays/holo-nixpkgs/hpos-holochain-api/README.md [hunk 2]
```diff
@@ -63,7 +63,7 @@ This endpoint is called to install/enable a hosted happ by passing the happ_id a
 ## Testing
 - ### Testing locally:
   - See that you are root of the `/hpos-holochain-api/` folder
-  - `npm install`
+  - `yarn install`
```
Comment (alastairong) [inline overlays/holo-nixpkgs/hpos-holochain-api/README.md:66]: PR #717 switches from yarn to npm. Which one are we trying to use? @robbiecarlton

File: overlays/holo-nixpkgs/hpos-holochain-api/src/index.js [hunk 3]
```diff
@@ -12,27 +12,10 @@ const { parsePreferences, isusageTimeInterval } = require('./utils')
 const { getAppIds, getReadOnlyPubKey } = require('./const')
 const { AdminWebsocket, AppWebsocket } = require('@holochain/conductor-api')
 
-// NB: `/hosted_happs` accepts `usageTimeInterval` as its only param - this value is passed to SL to calcuate the usage data for said time interval
-// usageTimeInterval = {
-//   durationUnit: String // accepted units: 'HOUR', 'DAY', 'WEEK', 'MONTH', 'YEAR
-//   amount: Int
-// }
```
[the remark this answers is not in the document]
Comment (robbiecarlton) [reply overlays/holo-nixpkgs/hpos-holochain-api/src/index.js:19]: For me, this kind of info properly belongs in the docs (which it is), not in the comments of the code. Comments in the code should be to explain things in the code that are not immediately obvious for people making changes to that code. If I'm using an API, I want to be able to look to the docs first to get information about how to use the api, and ideally I never have to read the code just to understand how to use the API.

It's not free to put comments in the code. It increases the energy required to read. One of our code style agreements is "keep the code base as small and tidy as possible", and removing this comment was in the spirit of that.

Of course, this is open to discussion.

File: overlays/holo-nixpkgs/hpos-holochain-api/src/index.js [hunk 4]
```diff
@@ -58,9 +41,7 @@ app.get('/hosted_happs', async (req, res) => {
       break
     }
 
-    // nb: servicelogger bandwidth payload is calculated in Bytes (not bits)
-    // and servicelogger cpu is calculated in microseconds (not seconds)
```
[the remark this answers is not in the document]
Comment (robbiecarlton) [reply overlays/holo-nixpkgs/hpos-holochain-api/src/index.js:62]: See the comment above. This information is rarely going to be useful to maintainers of this code. It's much more useful to consumers of the API, so it's better in the docs. I put it in there.

File: overlays/holo-nixpkgs/hpos-holochain-api/tests/index.test.js (hunk 1 above)
Comment (robbiecarlton) [reply overlays/holo-nixpkgs/hpos-holochain-api/tests/index.test.js:26]: Yes, will do 👍

File: overlays/holo-nixpkgs/hpos-holochain-api/README.md [hunk 5]
```diff
@@ -63,7 +63,7 @@ This endpoint is called to install/enable a hosted happ by passing the happ_id a
 ## Testing
```
[the remark this answers is not in the document]
Comment (robbiecarlton) [reply overlays/holo-nixpkgs/hpos-holochain-api/README.md:90]: Done

File: overlays/holo-nixpkgs/hpos-holochain-api/README.md (hunk 2 above)
Comment (robbiecarlton) [reply overlays/holo-nixpkgs/hpos-holochain-api/README.md:93]: I don't see anything about yarn or npm in PR #717, am I missing something?

This folder currently has a `yarn.lock` not a `package-lock.json`, and my understanding from a comment from @zo-el is that that was intentional. I may be wrong about this though.

Comment (JettTech) [reply overlays/holo-nixpkgs/hpos-holochain-api/README.md:93]: Yes, relating back to the PL conversation we had last Thurs, all holo repos need to be updated to use yarn due to the nix derivation needs.

File: overlays/holo-nixpkgs/hpos-holochain-api/src/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (JettTech) [reply overlays/holo-nixpkgs/hpos-holochain-api/src/index.js:19]: Keeping it in the README is clear enough.

**Review by JettTech — APPROVED** [review]
LGTM!

**Pull request merged**